### DISCUSSION TITLE
使用VS2022提示Jolt错误

### DISCUSSION THREAD

**Discussion by Suceru** [BoomingTech/Piccolo#336] [Q&A] [answered]
**Describe the bug | Bug描述**
```
严重性	代码	说明	项目	文件	行	禁止显示状态
错误	C1083	无法打开预编译头文件: “Jolt/Jolt.pch”: No such file or directory	F:\code\Vs2022\Piccolo-main\out\build\x64-Debug\Piccolo-main	F:\code\Vs2022\Piccolo-main\engine\3rdparty\JoltPhysics\Jolt\Physics\Collision\Shape\CylinderShape.cpp	4	
```

**Steps to reproduce | 如何复现**
1. 下载main.zip后进行解压，然后不使用cmake，用vs2022直接打开文件夹为main。
2. 找到`F:\code\Vs2022\Piccolo-main\engine\source\editor\source\main.cpp`设置为启动项目。
3. 点击启动，会提示上面错误。

**Additional context | 补充信息**
是否可以如同前面`Piccolo-games104-homework01-build`不用cmake，直接可以运行呢

**Comment by Ol6rin** [comment, marked as answer]
使用VS直接打开文件夹的方式其实也是利用了CMake，只是用的是VS中的内置CMake。如果不做任何配置，默认使用的Generator是Ninja。Windows上目前我们没有支持使用Ninja作为Generator的情况。
如果需要使用VS内置CMake编译，可以通过Project->CMake Settings for Piccolo中的高级设置
![image](https://user-images.githubusercontent.com/102927327/187686699-7597d26c-383d-4922-a94a-2a19d5dcd858.png)
![image](https://user-images.githubusercontent.com/1993283/187685687-2aab6d08-5824-45fb-9d69-c58d862c6313.png)
更改CMake generator为Visual Studio 17 2022 Win64，生成VS2022的工程编译。
![image](https://user-images.githubusercontent.com/1993283/187685802-6978a04a-7822-466e-a045-e2aed2dcb9fd.png)
另外，通过这种方式配置的工程直接调试启动会因为配置文件的相对路径问题找不到资源运行失败。可以手动运行顶层bin目录下的exe然后附加到进程调试。

WIndows下的最佳开发体验我们依然推荐先运行命令行通过外部CMake生成工程之后再用VS打开工程开发。

**Comment by Kylewlk** [comment]
这个是Jolt内部使用了预编译头文件，在Jolt.cmake中去掉就行预编译或者Ninja中禁用预编译头文件，就可以正常编译了。
![image](https://user-images.githubusercontent.com/22131472/222094338-f6658305-db66-4d0a-b5c9-ee76bfddfa5f.png)

 AND NOT "${CMAKE_GENERATOR}" STREQUAL "Ninja"
